### ISSUE TITLE
AWS::Logs::LogGroup: !GetAtt LogGroup.Arn returns the wrong value

### ISSUE THREAD

**Issue by scottbrown** [aws-cloudformation/aws-cloudformation-coverage-roadmap#206]
## 1. Title

AWS::Logs::LogGroup
Fn::GetAtt Arn

## 2. Scope of Request

The resource currently returns `Arn` through `GetAtt` but the value of this returned variable does not follow the ARN format posted [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazoncloudwatchlogs.html#amazoncloudwatchlogs-resources-for-iam-policies).  Currently it returns `arn:${Partition}:logs:${Region}:${Account}:log-group:${LogGroupName}:*`.  The extra `:*` appended to the return value means that it cannot be used in IAM policies or anywhere else that requires a clean CloudWatchLogs LogGroup ARN.

## 3. Expected Behaviour

`!GetAtt LogGroup.Arn` should return an ARN in the format `arn:${Partition}:logs:${Region}:${Account}:log-group:${LogGroupName}`

## 4. Suggested Test Cases

Given a `LogGroup` resource, use it in an IAM policy like so and run the policy simulator or a Lambda function to verify that logs are being written to the log group:
```json
Policy:
  Type: "AWS::IAM::Policy"
  Properties:
    PolicyName: "allow-lambda-logging"
    PolicyDocument:
      Version: "2012-10-17"
      Statement:
        - Effect: Allow
          Action:
            - logs:CreateLogStream
            - logs:PutLogEvents
          Resource: 
            - !GetAtt LogGroup.Arn
            - !Sub
              - "${Arn}:log-stream/*"
              - Arn: !GetAtt LogGroup.Arn
    Roles:
      - !Ref LambdaRole
```

## 5. Helpful Links

* CFN documentation showing the incorrect ARN format: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-loggroup.html#aws-resource-logs-loggroup-return-values
* Official ARN format: https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazoncloudwatchlogs.html#amazoncloudwatchlogs-resources-for-iam-policies

## 6. Category

Management

**Comment by AntonyPritchett** [comment]
Has this been fixed? I'm attempting to use GetAtt LogGroup.Arn as a target for EventBridge rule, but it doesn't like the :* that's appended

**Comment by asaushkin** [comment]
Wow! What's the weird behavior!

**Comment by PatMyron** [comment]
should be introduced as another separate attribute at this point to preserve backwards compatibility

**Comment by benbridts** [comment]
I agree with Pat, as the current behaviour is probably used in multiple IAM policies (where the :* is needed for PutLogEvents).

To get the version without the :* from the current output, you could use

```
!Select [0, !Split [':*', !GetAtt LogGroup.Arn ]]
```

**Comment by jk2l** [comment]
> I agree with Pat, as the current behaviour is probably used in multiple IAM policies (where the :* is needed for PutLogEvents).
> 
> To get the version without the :* from the current output, you could use
> 
> ```
> !Select [0, !Split [':*', !GetAtt LogGroup.Arn ]]
> ```

that is a very interesting way to work around a bug.

**Comment by sharuzzaman** [comment]
I stumbled upon this bug today when trying to create a LogGroup for VPNConnection. On the web UI, the result is just '*', which is the end part of the ARN for Cloudwatch Log.

Please create another attribute that does not have the ':*' at the back, and make our life and the Cloudformation template simpler.

Bug first reported in 2019, after 6 years (2025) the resource behaviour is still the same. Please improve your process AWS!

Thank you.

**Comment by jk2l** [comment]
> I stumbled upon this bug today when trying to create a LogGroup for VPNConnection. On the web UI, the result is just '*', which is the end part of the ARN for Cloudwatch Log.
> 
> Please create another attribute that does not have the ':*' at the back, and make our life and the Cloudformation template simpler.
> 
> Bug first reported in 2019, after 6 years (2025) the resource behaviour is still the same. Please improve your process AWS!
> 
> Thank you.

don't forget to thumb up the original post. it's how they track how important is a feature request i think

**Comment by sharuzzaman** [comment]
> > I stumbled upon this bug today when trying to create a LogGroup for VPNConnection. On the web UI, the result is just '*', which is the end part of the ARN for Cloudwatch Log.
> > Please create another attribute that does not have the ':*' at the back, and make our life and the Cloudformation template simpler.
> > Bug first reported in 2019, after 6 years (2025) the resource behaviour is still the same. Please improve your process AWS!
> > Thank you.
> 
> don't forget to thumb up the original post. it's how they track how important is a feature request i think

That's the first thing I will do if I found a bug report that is still not solved by AWS 😊
